### ISSUE TITLE
is it safe to assume this repo's setup same as original paper?

### ISSUE THREAD

**Issue by muiPomeranian** [lucidrains/reformer-pytorch#99]
Hi Lucid! thanks for the nice work !!

I have a quick question about the repo's setup. 

Can I assume that this repo's intention is try to reproduce the original paper's setup? 

Q1)
For enwik8,
I see that test and val data retrieval are not "Strictly" seperated. We randomly retrieve the 4096 sequence for train and val. It seems we might pick val sequence that have been used for train. Is there any reason? 
https://github.com/lucidrains/reformer-pytorch/blob/master/examples/enwik8_simple/train.py#L72

Q2)
According to the paper, we don't calc the self-attention of different buckets from <Q,K>. But according to this repo, 

self._attend_across_buckets = True
and seems we DO calc the self-attention of different buckets too
am I misunderstanding? 
https://github.com/lucidrains/reformer-pytorch/blob/master/reformer_pytorch/reformer_pytorch.py#L180

https://github.com/lucidrains/reformer-pytorch/blob/master/reformer_pytorch/reformer_pytorch.py#L354

Q3)
Seems we have scaleNorm option but we are using Layer Norm. Is this following as original paper?

Q4)
In case your basic set up is not meant to be following the original paper's setup, May you provide some recommendation which part I need to change to mock the original paper's result? 

Q5)
All the best and thanks a lot for this amazing work again !

**Comment by lucidrains** [comment]
Hello!

1. it is separate, since the validation text is sampled from another dataset https://github.com/lucidrains/reformer-pytorch/blob/master/examples/enwik8_simple/train.py#L82

2. the authors have a bunch of experimental settings not mentioned in the paper when I transcribed the code from their Trax repository. I have asked them about these settings, and they told me they were not necessary. I kept this in there in case any researchers wanted to play around with turning it on/off themselves.

3. scale norm is another simpler normalization alternative I found useful. the paper uses layer norm, so you can just keep this as `False`

4. If you omit most of the settings except for the dropouts and `ff_chunks` and `attn_chunks`, it will be what the paper has

5. Thanks! Hope you found it useful / educational

Q2)
According to the paper, we don't calc the self-attention of different buckets from <Q,K>. But according to this repo,

self._attend_across_buckets = True
and seems we DO calc the self-attention of different buckets too
am I misunderstanding?
https://github.com/lucidrains/reformer-pytorch/blob/master/reformer_pytorch/reformer_pytorch.py#L180

https://github.com/lucidrains/reformer-pytorch/blob/master/reformer_pytorch/reformer_pytorch.py#L354

Q3)
Seems we have scaleNorm option but we are using Layer Norm. Is this following as original paper?

Q4)
In case your basic set up is not meant to be following the original paper's setup, May you provide some recommendation which part I need to change to mock the original paper's result?

Q5)
All the best and thanks a lot for this amazing work again !
